### PR TITLE
MINOR: Reduce log levels for transactions_mixed_versions_test with 3.2 due to bug in that version

### PR DESCRIPTION
**Pull request by jolshan**
https://github.com/apache/kafka/commit/7496e6243410ca851f4b8502270cf4222bd89a2b fixed an error that caused an exception to be thrown on broker startup when debug logs were on. This made it to every version except 3.2. 

The Kraft upgrade tests totally turn off debug logs, but I think we only need to remove them for the broken version.

Note: this bug is also present in 3.1, but there is no logging on startup like in subsequent versions.

### CODE REVIEW

File: tests/kafkatest/tests/core/transactions_mixed_versions_test.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def test_transactions_mixed_versions(self, old_kafka_version, metadata_quorum=qu
         self.kafka.interbroker_security_protocol = security_protocol
         self.kafka.logs["kafka_data_1"]["collect_default"] = True
         self.kafka.logs["kafka_data_2"]["collect_default"] = True
-        self.kafka.logs["kafka_operational_logs_debug"]["collect_default"] = True
+        self.kafka.logs["kafka_operational_logs_debug"]["collect_default"] = if old_kafka_version == str(LATEST_3_2) False else True
```
Comment (chia7712) [inline tests/kafkatest/tests/core/transactions_mixed_versions_test.py]: It seems 3.1.2 has same issue. see https://github.com/apache/kafka/blob/3.1.2/metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistration.java

Maybe `LATEST_3_1` should be included too?

**Review by chia7712 — COMMENTED** [review]
@jolshan thanks for this patch. BTW, does `test_transactions_upgrade` have similar story?

https://github.com/apache/kafka/blob/trunk/tests/kafkatest/tests/core/transactions_upgrade_test.py#L220

**Comment by jolshan** [comment]
I have not seen it on other versions. I thought I ran the tests, but looks like I didn't. I will update the PR when I see the results.

**Comment by jolshan** [comment]
I ran the transactions_upgrade_test without any changes and everything passed.

I ran the mixed_version_test but it is failing due to https://issues.apache.org/jira/browse/KAFKA-17250 :(

**Comment by chia7712** [comment]
> I ran the mixed_version_test but it is failing due to https://issues.apache.org/jira/browse/KAFKA-17250 :(

@jolshan Could you please take a look at my previous comment: https://github.com/apache/kafka/pull/16235#discussion_r1672760435?

**Comment by jolshan** [comment]
Looks like the other PR is merged. I will try rerunning the tests now

**Comment by jolshan** [comment]
All the tests pass now. We don't need for 3.1 since it doesn't have this line (I traced back all the components of delta and it eventually gets to a BrokerRegistration):

3.2:
https://github.com/apache/kafka/blob/e4ca066680296ea29d443efb626baecc837083f6/core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala#L274

3.1:
https://github.com/apache/kafka/blob/2b57b38f9306ac73f18d94d55c0158530fd444df/core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala#L254

**Review by dajac — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
I will backport this to 3.9 for consistency before backporting #17067

**Pull request merged**